File: LLM-Translations/GPT-4/avatar/Java/C/atcoder_AGC006_B.c
#include <stdio.h>

int main() {
    int n, x, max;
    scanf("%d", &n);
    scanf("%d", &x);

    max = n * 2 - 1;

    if (x == 1 || x == max) {
        printf("No\n");
        return 0;
    }

    printf("Yes\n");

    for (int i = x + n - 1; i < x + n + max - 1; i++) {
        printf("%d\n", i % max + 1);
    }

    return 0;
}

// //End of Code.

File: LLM-Translations/GPT-4/avatar/Java/Go/atcoder_AGC006_B.go

package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	var n, x int
	fmt.Scan(&n, &x)
	print(n, x)
}

func print(n int, x int) {
	max := n*2 - 1
	if x == 1 || x == max {
		fmt.Println("No")
		return
	}
	fmt.Println("Yes")
	var builder strings.Builder
	for i := x + n - 1; i < x+n+max-1; i++ {
		builder.WriteString(strconv.Itoa(i%max+1) + "\n")
	}
	fmt.Println(builder.String())
}

// //End of Code.
